docs/development/workflow/maintainer_workflow.rst

```diff
@@ -7,105 +7,108 @@ Workflow for Maintainers
 This page is for maintainers |emdash| those of us who merge our own or other
 peoples' changes into the upstream repository.
 
-Being as how you're a maintainer, you are completely on top of the basic stuff
+Being a maintainer with write access, you are expected to be on top of the basic stuff
 in :ref:`development-workflow`.
 
-=======================================================
-Integrating changes via the web interface (recommended)
-=======================================================
+=========================================
+Integrating changes via the web interface
+=========================================
 
-Whenever possible, merge pull requests automatically via the pull request manager on GitHub. Merging should only be done manually if there is a really good reason to do this!
+Pull requests should always be merged via GitHub.
 
-Make sure that pull requests do not contain a messy history with merges, etc. If this is the case, then follow the manual instructions, and make sure the fork is rebased to tidy the history before committing.
+Make sure that pull requests do not contain a messy history with merges, etc.
+If this is the case, ask the author to squash the commits or
+:ref:`take over the pull request <maintainer-pr-takeover>`.
+If necessary, you may also :ref:`request a rebase <rebase>`.
```

Comment:
I think it might be better to encourage the maintainers to do the squashing and rebasing, but that would be beyond the scope of this pull request.

Reply:
Usually, when I ask users to rebase, I offer to do it for them, because it can be a hassle. On the other hand, maintainer time is precious and it is good for contributors to be able to do it, since once they know how, they can do it substantially faster than I can (since I have to add the remote, etc.). Anyway, as you note, bit out of scope here.